cpp/CMakeLists.txt

```diff
@@ -23,636 +23,595 @@ cmake_minimum_required(VERSION 3.14...3.17 FATAL_ERROR)
-
-if(BUILD_BENCHMARKS)
-
-  include(ConfigureGoogleBenchmark)
-
-  if(GBENCH_FOUND)
-    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
-    include_directories(${GBENCH_INCLUDE_DIR})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
-  else()
-    message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found: automated tests are disabled.")
-  endif(GBENCH_FOUND)
-
-endif(BUILD_BENCHMARKS)
-
-message(STATUS "BENCHMARK_LIST set to: ${BENCHMARK_LIST}")
-
-###################################################################################################
-# - include paths ---------------------------------------------------------------------------------
+    target_compile_definitions(cudf PUBLIC CUDA_API_PER_THREAD_DEFAULT_STREAM)
+endif()
 
-include_directories("${THRUST_INCLUDE_DIR}"
-                    "${JITIFY_INCLUDE_DIR}"
-                    "${LIBCUDACXX_INCLUDE_DIR}")
+# Disable NVTX if necessary
+if(NOT USE_NVTX)
+    target_compile_definitions(cudf PUBLIC NVTX_DISABLE)
+endif()
 
-if(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES)
-	include_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
-endif(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES)
+# Define spdlog level
+target_compile_definitions(cudf PUBLIC "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${RMM_LOGGING_LEVEL}")
 
-include_directories("${CMAKE_BINARY_DIR}/include"
-                    "${CMAKE_SOURCE_DIR}/include"
-                    "${CMAKE_SOURCE_DIR}/src"
-                    "${ARROW_INCLUDE_DIR}"
-                    "${ZLIB_INCLUDE_DIRS}"
-                    "${Boost_INCLUDE_DIRS}"
-                    "${RMM_INCLUDE}"
-                    "${DLPACK_INCLUDE}")
+# Compile stringified JIT sources first
+add_dependencies(cudf stringify_run)
 
-if(CONDA_INCLUDE_DIRS)
-    include_directories("${CONDA_INCLUDE_DIRS}")
-endif(CONDA_INCLUDE_DIRS)
+# Specify the target module library dependencies
+target_link_libraries(cudf
+           PUBLIC ZLIB::ZLIB
+                  Boost::filesystem
+                  ${ARROW_LIBRARIES}
+                  cudf::Thrust
+                  rmm::rmm)
 
-###################################################################################################
-# - library paths ---------------------------------------------------------------------------------
+if(CUDA_STATIC_RUNTIME)
+    target_link_libraries(cudf PUBLIC CUDA::nvrtc CUDA::cudart_static CUDA::cuda_driver)
+else()
+    target_link_libraries(cudf PUBLIC CUDA::nvrtc CUDA::cudart CUDA::cuda_driver)
+endif()
 
-link_directories("${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}" # CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES is an undocumented/unsupported variable containing the link directories for nvcc
-                 "${CMAKE_BINARY_DIR}/lib"
-                 "${CMAKE_BINARY_DIR}"
-                 "${GTEST_LIBRARY_DIR}")
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
I'm -1 on doing this only for non-release builds. In theory we could eventually hit this same problem in release builds as we continue to expand the library and device architectures we support.

I know you mentioned this causing issues downstream in certain situations. Should we try to fix/address those issues instead?

Reply:
Yeah makes sense, I'll revert it. This is mostly cosmetic, it just causes gcc to print a warning that devs might think is something to worry about.